Python/phase1.py

```diff
@@ -0,0 +1,217 @@
+    return data
+
+# group the victims by room and sum up.
+def get_new_loc(data):
+    location_df = pd.json_normalize(data ,"locations")  
+    new_loc_df = location_df.groupby(['area_id'])[['victims.critical', 'victims.non_critical']].sum().reset_index() 
+    return new_loc_df
+
+# calculate remaining yellow victims in the game
+def calculate_remain_yellow_victims(new_loc_df, new_df):
+    total_yellow_victim = sum(new_loc_df['victims.critical'])    
+    new_df['remain_yellow_victim'] = np.where(new_df['data_color'] == 'Yellow', 1, 0)
+    new_df['remain_yellow_victim'] = total_yellow_victim - new_df['remain_yellow_victim'].cumsum()  
+    return new_df
+
+# calculate current room green and yellow victimes
+def get_cur_room_victims(new_df, new_loc_df):  
+    new_df['cur_room'] = new_df['data_entered_area_id'].ffill()    
+    new_df = pd.merge(new_df, new_loc_df, left_on='cur_room', right_on='area_id', how='left')    
+    new_df[['cur_room_yellow_victims', 'cur_room_green_victims']] = new_df[['victims.critical', 'victims.non_critical']].fillna(value=0)
+    return new_df    
+
+# calculate if next room entered has yellow vicims or not    
+def next_room_has_yellow_victim(new_df, new_loc_df):   
+    room_yellow_victim_df =  new_loc_df[['area_id']]  
+    room_yellow_victim_df['next_room_has_yellow_victim'] = new_loc_df['victims.critical']==1.0   
+    new_df = pd.merge(new_df, room_yellow_victim_df, left_on='next_room', right_on='area_id', how='left')    
+    new_df['next_room_has_yellow_victim'] = new_df['next_room_has_yellow_victim'].fillna(value=False)    
+    return new_df
+
+# read in the condition file that contains Ss information
+def read_condition(condition_path):
+    condition_df = pd.read_csv(condition_path)
+    return condition_df      
+
+# join the tables to get necessary information
+def join_condition(condition_df, fname, subject_id, new_df):
+    mission_complex_df = condition_df[['condition_id', 'condition_within_Ss']]    
+    study_id = fname[11:26] 
+    condition_id = fname[27:46]   
+    trial_id = fname[47:62]
+    new_df['study_id'] = study_id
+    new_df['condition_id'] = condition_id    
+    new_df['trial_id'] = trial_id    
+    new_df['subject_id'] = subject_id      
+    new_df = pd.merge(new_df, mission_complex_df, left_on='condition_id', right_on='condition_id', how='left')    
+    training_df = condition_df[['condition_id', 'condition_between_Ss']]   
+    new_df = pd.merge(new_df, training_df, left_on='condition_id', right_on='condition_id', how='left') 
+    return new_df
+
+# pre-process the survey file
+def get_survey_table(survey_path):
+    survey_df = pd.read_csv(survey_path)    
+    survey_df = survey_df.drop([0, 1]).reset_index(drop=True)    
+    survey_df=survey_df[['Q2', 'Q5_1','Q5_2','Q5_3','Q5_4','Q5_5','Q5_6','Q5_7','Q5_8','Q5_9','Q5_10']]    
+        
+    survey_df['Q5_1'] = survey_df['Q5_1'].apply(lambda x: x[:-4] if len(x)>3 else x)
```

Comment:
Can you put lines 179-188 in a loop?

Reply:
Hi Cynthia, 
The below code could work for Pablos suggestion. Feel free to use. 
```
    sat_cols = ['Q2', 'Q5_1','Q5_2','Q5_3','Q5_4','Q5_5','Q5_6','Q5_7','Q5_8','Q5_9','Q5_10']

    survey_df=survey_df[sat_cols]

    # Cleaning columns
    # Start at 1 as we want to skip Q2
    i = 1
    while i < len(sat_cols):
        survey_df[sat_cols[i]] = (survey_df[sat_cols[i]].apply(lambda x: x[:-4] if len(str(x))>3 else x))

    i += 1

    return df
```

Reply:
> Hi Cynthia,
> The below code could work for Pablos suggestion. Feel free to use.
> 
> ```
>     sat_cols = ['Q2', 'Q5_1','Q5_2','Q5_3','Q5_4','Q5_5','Q5_6','Q5_7','Q5_8','Q5_9','Q5_10']
> 
>     survey_df=survey_df[sat_cols]
> 
>     # Cleaning columns
>     # Start at 1 as we want to skip Q2
>     i = 1
>     while i < len(sat_cols):
>         survey_df[sat_cols[i]] = (survey_df[sat_cols[i]].apply(lambda x: x[:-4] if len(str(x))>3 else x))
> 
>     i += 1
> 
>     return df
> ```

Thanks, I know how to code, I found a more efficient way. Please feel free to leave more comments.